apis/v1alpha1/gateway_types.go

```diff
@@ -483,47 +485,125 @@ type GatewayStatus struct {
+// GatewayConditionReason defines the set of reasons that explain
+// why a particular Gateway condition type has been raised.
+type GatewayConditionReason string
 
+const (
 	// ConditionGatewayNotScheduled indicates that the Gateway has not been
 	// scheduled.
-	ConditionGatewayNotScheduled GatewayConditionType = "GatewayNotScheduled"
+	ConditionGatewayNotScheduled GatewayConditionType = "NotScheduled"
 
 	// ConditionListenersNotReady indicates that at least one of the specified
 	// listeners is not ready. If this condition has a status of True, a more
 	// detailed ListenerCondition should be present in the corresponding
 	// ListenerStatus.
 	ConditionListenersNotReady GatewayConditionType = "ListenersNotReady"
+)
 
-	// ConditionInvalidListeners indicates that at least one of the specified
-	// listeners is invalid. If this condition has a status of True, a more
-	// detailed ListenerCondition should be present in the corresponding
-	// ListenerStatus.
-	ConditionInvalidListeners GatewayConditionType = "InvalidListeners"
+const (
+	// GatewayConditionScheduled indicates whether the controller
+	// managing the Gateway has scheduled the Gateway to the
+	// underlying network infrastructure.
+	//
+	// Valid reasons for this condition to be false are:
+	//
+	// * "NotReconciled"
+	// * "NamespaceForbidden"
+	// * "NoSuchGatewayClass"
+	// * "NoResources"
+	GatewayConditionScheduled GatewayConditionType = "Scheduled"
+
+	// GatewayReasonNotReconciled is used when the Gateway is
+	// not scheduled because it recently been created and no
+	// controller has reconciled it yet.
+	GatewayReasonNotReconciled GatewayConditionReason = "NotReconciled"
+
+	// GatewayReasonNamespaceForbidden is used when the Gateway is
+	// not scheduled because, while the Gateway is visible to
+	// the managing controller, the controller's configuration
+	// forbids is from scheduling objects from that namespace.
+	GatewayReasonNamespaceForbidden GatewayConditionReason = "NamespaceForbidden"
+
+	// GatewayReasonNoSuchGatewayClass is used when the Gateway is
+	// not scheduled because there is no controller that recognizes
+	// the gateway class name. This reason should only be set by
+	// a controller that has cluster-wide visibility of all the
+	// installed gateway classes.
+	GatewayReasonNoSuchGatewayClass GatewayConditionReason = "NoSuchGatewayClass"
+
+	// GatewayReasonNoResources is used when the Gateway is
+	// not scheduled because no infrastructure resources are
+	// available for this Gateway.
+	GatewayReasonNoResources GatewayConditionReason = "NoResources"
+)
 
-	// ConditionInvalidAddress indicates one or more of the
-	// Gateway's Addresses is invalid or could not be assigned.
-	ConditionInvalidAddress GatewayConditionType = "InvalidAddress"
+const (
+	// GatewayConditionReady indicates whether the Gateway is able
+	// to serve traffic. Note that this does not indicate that the
+	// Gateway configuration is current or even complete.
```

Comment:
> this does not indicate that the Gateway configuration is current or even complete.

What does that mean? Is this meant to indicate that a Gateway is ready to receive traffic but may not represent the most up to date configuration here? Or that a Gateway is ready but the configuration here is invalid?

Reply:
You can be ready but not current (if the spec changed), but also partial in some senses. For example, if you can't resolve routes or you can resolve the CA secrets for some of the routes.